bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/tracer/TracingUtils.java

```diff
@@ -47,24 +46,25 @@ private TracingUtils() {
      * @param isClient        true if the starting span is a client
      */
     public static void startObservation(ObserverContext observerContext, boolean isClient) {
-        BSpan span = new BSpan(observerContext, isClient);
-        span.setServiceName(observerContext.getServiceName());
-
-        span.setOperationName(observerContext.getOperationName());
-        if (isClient) {
-            observerContext.addProperty(PROPERTY_TRACE_PROPERTIES, span.getProperties());
+        BSpan span;
+        if (observerContext.getParent() != null) {
+            BSpan parentSpan = observerContext.getParent().getSpan();
+            span = BSpan.start(parentSpan, observerContext.getServiceName(), observerContext.getOperationName(),
```

Comment:
lets move `observerContext.getServiceName()` and `observerContext.getOperationName()` to variables.

Reply:
Fixed in 6912281